src/agent/src/image_rpc.rs

```diff
@@ -136,9 +188,15 @@ impl ImageService {
             verify_cid(cid)?;
         }
 
+        if aa_kbc_params != "" && !aa_started {
+            Self::init_attestation_agent();
+            let mut agent_config = AGENT_CONFIG.write().await;
```

Comment:
@sameo we see this lock failing to be acquired with shim pull, would you have an idea why that happens?

Reply:
New version sidesteps the agent config by adding an AtomicBool as a field in the image service.